### ISSUE TITLE
Bad PCC for matmul trace 128, 1024, 768

### ISSUE THREAD

**Issue by nsorabaTT** [tenstorrent/tt-metal#32584]
I encountered bad PCC that gets progressively worse as the "K" dim is increased. Running the below snippet shows a PCC of 0.99, 0.98 and 0.95 for the 3 cases.

These examples are a small subset of the traces in https://github.com/tenstorrent/tt-metal/pull/32295 which are also failing PCC check.

```python
import torch
import ttnn
import pytest
from tests.ttnn.utils_for_testing import assert_with_pcc


@pytest.mark.parametrize("mkn", [(128, 512, 768), (128, 1024, 768), (128, 2048, 768)])
def test_matmul_without_transpose_show_bad_pcc(device, mkn):
    torch.manual_seed(0)

    m, k, n = mkn

    # Create input tensors
    torch_input_tensor_a = torch.rand((m, k), dtype=torch.bfloat16)
    torch_input_tensor_b = torch.rand((k, n), dtype=torch.bfloat16)

    torch_output_tensor = torch.matmul(torch_input_tensor_a, torch_input_tensor_b)

    input_tensor_a = ttnn.from_torch(torch_input_tensor_a, layout=ttnn.TILE_LAYOUT, device=device)
    input_tensor_b = ttnn.from_torch(torch_input_tensor_b, layout=ttnn.TILE_LAYOUT, device=device)

    output = ttnn.matmul(input_tensor_a, input_tensor_b)
    output = ttnn.to_torch(output)

    assert_with_pcc(torch_output_tensor, output, 0.999)
```

**Comment by nsorabaTT** [comment]
Using `fp32_dest_acc_en` and `packer_l1_acc` improved the PCC to over 0.99.

@bbradelTT do we have an issue for enabling fp32 accumulation by default for matmuls? I would like to link that here and close this one.

**Comment by bbradelTT** [comment]
I haven't created one yet because I was waiting for feedback first. I'll use this one.

**Comment by bbradelTT** [comment]
Issue where accuracy was increased for fp32: https://github.com/tenstorrent/tt-metal/issues/16366 and it's PR https://github.com/tenstorrent/tt-metal/pull/16567
